controller/admin_transactions_controller.php

```diff
@@ -296,11 +307,98 @@ private function do_action($action, $args)
 				}
 				// Clear $action status
 				$action = '';
+			break;
+			case 'add':
+				$errors = array();
+
+				$transaction_data = array(
+					'MT_ANONYMOUS'          => $this->request->is_set('u'),
+					'MT_USERNAME'           => $this->request->variable('username', '', true),
+					'MT_FIRST_NAME'         => $this->request->variable('first_name', '', true),
+					'MT_LAST_NAME'          => $this->request->variable('last_name', '', true),
+					'MT_PAYER_EMAIL'        => $this->request->variable('payer_email', '', true),
+					'MT_RESIDENCE_COUNTRY'  => $this->request->variable('residence_country', ''),
+					'MT_MC_GROSS'           => $this->request->variable('mc_gross', (float) 0),
+					'MT_MC_CURRENCY'        => $this->request->variable('mc_currency', ''),
+					'MT_MC_FEE'             => $this->request->variable('mc_fee', (float) 0),
+					'MT_PAYMENT_DATE_YEAR'  => $this->request->variable('payment_date_year', (int) $this->user->format_date(time(), 'Y')),
+					'MT_PAYMENT_DATE_MONTH' => $this->request->variable('payment_date_month', (int) $this->user->format_date(time(), 'n')),
+					'MT_PAYMENT_DATE_DAY'   => $this->request->variable('payment_date_day', (int) $this->user->format_date(time(), 'j')),
+					'MT_PAYMENT_TIME'       => $this->request->variable('payment_time', $this->user->format_date(time(), 'H:i:s')),
+					'MT_MEMO'               => $this->request->variable('memo', '', true),
+				);
+
+				if ($this->request->is_set_post('submit'))
+				{
+					try
+					{
+						$data_ary = $this->build_data_ary($transaction_data);
+
+						$this->ppde_actions->log_to_db($data_ary);
+						$this->ppde_actions->update_overview_stats();
+						$this->ppde_actions->update_raised_amount();
+
+						if ($this->ppde_actions->get_donor_is_member())
```

Comment:
`get_donor_is_member()` will not work in this context because the property `$this->donor_is_member` is not set in the core_actions class.

I'll refactor (again 🤣 ) this part. In the meantime, it's better to check if `$data_ary['user_id'] != ANONYMOUS`

Reply:
Grrr... This not only this action which doesn't work.
To be able to use methods from core actions, you must sent a couple of properties/data.

After the `log_to_db()`, use
```
$this->ppde_actions->set_transaction_data($transaction_data);
$this->ppde_actions->is_donor_is_member();
```
This will send all data required by core actions, then you will be able to use donor_is_member or group_user_add...